scripts/builtin/decisionTreePredict.dml

```diff
@@ -0,0 +1,216 @@
+    print ("No such strategy" + strategy)
+    Y = matrix("0", rows=0, cols=0)
+  }
+}
+
+predict_TT = function (
+    Matrix[Double] M,
+    Matrix[Double] X
+) return (Matrix[Double] Y){
+
+  Y = matrix(0, rows=1, cols=nrow(X))
+  n = ncol(M)
+  tree_depth = ceiling(log(7,2)) # max depth of complete binary tree
+  [N_L, N_R, N_F, N_T, N_C] = createNodeTensors(M)
+
+  for (k in 1:nrow(X)){
+    # iterate over every sample in X matrix
+    sample = X[k,]
+    current_node = 1
+    cnt = 1
+    while (cnt < tree_depth){
+      feature_id = as.scalar(N_F[1, current_node])
+      feature = as.scalar(sample[,feature_id]) # select feature from sample data
+      threshold = as.scalar(N_T[1, current_node])
+
+      if (feature < threshold){
+        # move on to left child node
+        next_node = as.scalar(N_L[1, current_node])
+      } else {
+        # move on to right child node
+        next_node = as.scalar(N_R[1, current_node])
+      }
+      current_node = next_node
+      cnt +=1
+    }
+
+    class = M[4, current_node]
+    Y[1, k] = class
+  }
+}
+
+createNodeTensors = function(
+  Matrix[Double] M
+  ) return (
+    Matrix[Double] N_L,
+    Matrix[Double] N_R,
+    Matrix[Double] N_F,
+    Matrix[Double] N_T,
+    Matrix[Double] N_C
+    ){
+      N = M[1,] # all tree nodes
+      I = M[2,] # list of node offsets to their left children
+
+      N_L = matrix(0, rows=1, cols=0)
+      N_R = matrix(0, rows=1, cols=0)
+      N_F = matrix(0, rows=1, cols=0)
+      N_T = matrix(0, rows=1, cols=0)
+
+      for (i in 1:ncol(N)){
+        # if the node is an internal node, add its left child to the N_L tensor
+        if (as.scalar(I[1,i]) != 0){
+          offset = as.scalar(I[1, i])
+          N_L = cbind(N_L, N[1, i+offset])
+        } else {
+          N_L = cbind(N_L, as.matrix(i))
+        }
+      }
+
+      for (i in 1:ncol(N)){
+        # if the node is an internal node, add its right child to the N_R tensor
+
+        if (as.scalar(I[1,i]) != 0){
+          offset = as.scalar(I[1, i])
+          leftChild = as.scalar(N[1, i+offset])
+          rightChild = leftChild + 1
+
+          if (as.scalar(I[1, leftChild]) == 0 & as.scalar(I[1, rightChild]) != 0 ){
+            rightChild = i
+          }
+          N_R = cbind(N_R, N[1, rightChild])
+        
+        } else {
+          N_R = cbind(N_R, as.matrix(i))
+        }
+      }
+
+      for (i in 1:ncol(N)){
+        # if the node is an internal node, add index of the feature it evaluates
+        if (as.scalar(M[3,i]) != 0){
+          N_F = cbind(N_F, M[3,i])
+        
+        } else {
+          N_F = cbind(N_F, as.matrix(1))
+        }
+      }
+
+      for (i in 1:ncol(N)){
+        # if the node is an internal node, add the threshold of the feature it evaluates
+        if (as.scalar(M[6,i]) != 0){
+          N_T = cbind(N_T, M[6,i])
+        
+        } else {
+          N_T = cbind(N_T, as.matrix(0))
+        }
+      }
+
+    unique_classes = matrix(0, rows=1, cols=1)
```

Comment:
your indentation suddenly change here, please make the indentaton two spaces inside the function as a base.